### PR TITLE
[fix-#14012] add throwable Converter

### PR DESCRIPTION
**Pull request by fuchanghai**
## Purpose of the pull request
this close:
- #14012 
## Brief change log

add throwable Converter

after change:

image

image

## Verify this pull request

This change added tests and can be verified as follows:

SensitiveThrowableConverterTest#testPwdLogMsgConverter

### CODE REVIEW

**Comment by fuchanghai** [comment]
@caishunfeng @ruanwenjun @SbloodyS plz run CI

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/log/SensitiveThrowableConverter.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
+package org.apache.dolphinscheduler.common.log;
```
Comment (SbloodyS) [inline dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/log/SensitiveThrowableConverter.java:18]: Missing license header.

Comment (fuchanghai) [reply dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/log/SensitiveThrowableConverter.java:18]: done

**Review by Radeity — COMMENTED** [review]
I find that there are many duplicated codes with existed `SensitiveDataConverter`, can we abstract a class like `SensitiveConverter` and make code cleaner? 

BTW, may i ask can we override `convert` method to mask throwable logs, either?  If we can, this abstract work will be easier.

**Comment by fuchanghai** [comment]
> I find that there are many duplicated codes with existed `SensitiveDataConverter`, can we abstract a class like `SensitiveConverter` and make code cleaner?
> 
> BTW, may i ask can we override `convert` method to mask throwable logs, either? If we can, this abstract work will be easier.

thanks for you suggestion, I'll fix it later
BTW these two converters have inherited other classes, I can only define one interface

**Comment by rickchengx** [comment]
I think the following files also need to be updated
* the `logback-spring.xml` in `dolphinscheduler-worker` and `dolphinscheduler-standalone-server`
* the `logback.xml` in `dolphinscheduler-master`

**Comment by ruanwenjun** [comment]
I am +0 to this PR.
If someone needed to mask the sensitive message, they can write their own ThrowableProxyConverter. DS shouldn't need to consider a lot of these features.

**Comment by fuchanghai** [comment]
> I am +0 to this PR. If someone needed to mask the sensitive message, they can write their own ThrowableProxyConverter. DS shouldn't need to consider a lot of these features.

If one wants to hide IP information, SensitiveConverter can only hide the normal output information at this time. When an UnknownHostException is thrown, the IP information cannot be hidden. This is why I implemented ThrowableProxyConverter.

**Comment by ruanwenjun** [comment]
> > I am +0 to this PR. If someone needed to mask the sensitive message, they can write their own ThrowableProxyConverter. DS shouldn't need to consider a lot of these features.
> 
> If one wants to hide IP information, SensitiveConverter can only hide the normal output information at this time. When an UnknownHostException is thrown, the IP information cannot be hidden. This is why I implemented ThrowableProxyConverter.

I mean user can implement their own `ThrowableProxyConverter `, if someone want to mask username/password/personnumber/ip/publickKey/proxyIp, they will submit a lot of pr to change these...... this is not our core feature.

**Comment by fuchanghai** [comment]
> > > I am +0 to this PR. If someone needed to mask the sensitive message, they can write their own ThrowableProxyConverter. DS shouldn't need to consider a lot of these features.
> > 
> > 
> > If one wants to hide IP information, SensitiveConverter can only hide the normal output information at this time. When an UnknownHostException is thrown, the IP information cannot be hidden. This is why I implemented ThrowableProxyConverter.
> 
> I mean user can implement their own `ThrowableProxyConverter `, if someone want to mask username/password/personnumber/ip/publickKey/proxyIp, they will submit a lot of pr to change these...... this is not our core feature.

I know what you're worried about, so I'm going to make these rules configurable
you can see #13993 WDYT? @SbloodyS @Radeity

**Pull request closed without merging**